File: C++/Arrays/Insertionsort.cpp
#include <iostream>
using namespace std;

int main(){
    int n=7;
    int arr[7]={10,1,7,4,8,2,11};

    for(int i=0;i<n;i++){//Rounds
        int temp=arr[i];

        int j=i-1;
        for(;j>=0;j--){
            if(arr[j]>temp){
                //shift
                arr[j+1]=arr[j];
            }
            else{
                break;
            }
        }
        arr[j+1]=temp;
    }
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    return 0;
}

File: Java/basics/Insertionsort.java
import java.util.Arrays;
class Insertionsort{
    static void insertionsort(int[] arr){
        for(int i = 0;i<=arr.length-2;i++){
            for(int j = i+1; j>0 && arr[j-1]>arr[j];j--){
                swap(arr, j, j-1);
            }
        }
    }
    static void swap (int[] arr, int a, int b){
        int temp = arr[a];
        arr[a]= arr[b];
        arr[b]= temp;
    }
    public static void main(String[] args) {
        int[] nums = {5,4,3,2,1};
        insertionsort(nums);
        System.out.println(Arrays.toString(nums));
    }
}